src/routes/blossom.rs

```diff
@@ -265,6 +306,77 @@ async fn mirror(
     .await
 }
 
+#[rocket::post("/mirror-suggestions", data = "<req>", format = "json")]
```

Comment:
This should be built on the UI in ui_src, there is nothing to do on the backend we already support the /mirror request